features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,582 @@
+import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";
+
+import EmpState from "../../containers/EmpState";
+import DvmState from "../../containers/DvmState";
+import Token from "../../containers/Token";
+import EmpSponsors from "../../containers/EmpSponsors";
+import EmpContract from "../../containers/EmpContract";
+import Collateral from "../../containers/Collateral";
+import PriceFeed from "../../containers/PriceFeed";
+import Etherscan from "../../containers/Etherscan";
+
+const Label = styled.span`
+  color: #999999;
+`;
+
+const Status = styled(Typography)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const PositionDialog = styled.div`
+  width: 100%;
+  border-color: white;
+  padding-left: 15px;
+  padding-right: 15px;
+  padding-bottom: 20px;
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+interface DialogProps {
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const PositionActionsDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+  const { latestPrice } = PriceFeed.useContainer();
+  const {
+    symbol: tokenSymbol,
+    balance: tokenBalance,
+    allowance: tokenAllowance,
+    setMaxAllowance: setMaxTokenAllowance,
+  } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { activeSponsors } = EmpSponsors.useContainer();
+
+  const [dialogTabIndex, setDialogTabIndex] = useState<string | null>(
+    "deposit"
+  );
+  const [collateralToDeposit, setCollateralToDeposit] = useState<string>("");
+  const [minCollPerToken, setMinCollPerToken] = useState<string>("");
+  const [maxCollPerToken, setMaxCollPerToken] = useState<string>("");
+  const [maxTokensToLiquidate, setMaxTokensToLiquidate] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    priceIdentifier: priceId,
+    collateralRequirement: collReq,
+    currentTime,
+  } = empState;
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string | null
+  ) => {
+    setDialogTabIndex(newAlignment);
+  };
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String | null) => {
+    if (!x) return "N/A";
```

Comment:
is checking `!x` ALWAYS equivalent to `x !== null`?

Reply:
I think so? I'm not 100% sure with typescript. I changed it to be consistent with the rest of the code base for this function.

Reply:
I think it's equivalent only because there is no valid string that's falsy, so null is the only possible value for that type that could be treated as false (but I could be wrong!).